test/autorest/headergroup/headergroup_test.go

```diff
@@ -39,12 +39,526 @@ func deepEqualOrFatal(t *testing.T, result interface{}, expected interface{}) {
+		t.Fatalf("ParamProtectedKey: %v", err)
+	}
+	expected := &headergroup.HeaderParamProtectedKeyResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestHeaderParamString(t *testing.T) {
+	client := getHeaderClient(t)
+	result, err := client.ParamString(context.Background(), "valid", "The quick brown fox jumps over the lazy dog")
+	if err != nil {
+		t.Fatalf("ParamString: %v", err)
+	}
+	expected := &headergroup.HeaderParamStringResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+
+	// result, err = client.ParamString(context.Background(), "null", "")
+	// if err != nil {
+	// 	t.Fatalf("ParamString: %v", err)
+	// }
+	// expected = &headergroup.HeaderParamStringResponse{
+	// 	StatusCode: http.StatusOK,
+	// }
+	// deepEqualOrFatal(t, result, expected)
+
+	result, err = client.ParamString(context.Background(), "empty", "")
+	if err != nil {
+		t.Fatalf("ParamString: %v", err)
+	}
+	expected = &headergroup.HeaderParamStringResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+// TODO check why we dont check for the value returned in all of the tests below this comment
+func TestHeaderResponseBool(t *testing.T) {
+	client := getHeaderClient(t)
+	result, err := client.ResponseBool(context.Background(), "true")
+	if err != nil {
+		t.Fatalf("ResponseBool: %v", err)
+	}
+	expected := &headergroup.HeaderResponseBoolResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+
+	result, err = client.ResponseBool(context.Background(), "false")
+	if err != nil {
+		t.Fatalf("ResponseBool: %v", err)
+	}
+	expected = &headergroup.HeaderResponseBoolResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestHeaderResponseByte(t *testing.T) {
+	client := getHeaderClient(t)
+	result, err := client.ResponseByte(context.Background(), "valid")
+	if err != nil {
+		t.Fatalf("ResponseByte: %v", err)
+	}
+	expected := &headergroup.HeaderResponseByteResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestHeaderResponseDate(t *testing.T) {
+	client := getHeaderClient(t)
+	result, err := client.ResponseDate(context.Background(), "valid")
+	if err != nil {
+		t.Fatalf("ResponseDate: %v", err)
+	}
+	expected := &headergroup.HeaderResponseDateResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+
+	result, err = client.ResponseDate(context.Background(), "min")
+	if err != nil {
+		t.Fatalf("ResponseDate: %v", err)
+	}
+	expected = &headergroup.HeaderResponseDateResponse{
+		StatusCode: http.StatusOK,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestHeaderResponseDatetime(t *testing.T) {
+	client := getHeaderClient(t)
+	result, err := client.ResponseDatetime(context.Background(), "valid")
+	if err != nil {
+		t.Fatalf("ResponseDatetime: %v", err)
+	}
+	expected := &headergroup.HeaderResponseDatetimeResponse{
```

Comment:
The models for operations with header responses should have them baked in.  E.g. based on the swagger, `HeaderResponseDatetimeResponse` should have a field named `Value` that's a `time.Time`.

Reply:
Maybe this is already tracked in the work item you opened the other day.

Reply:
Yes I agree it should have them baked it, I dont know why it's not happening here. But it is listed the #270